packages/core/lib/debug/vscode.js

```diff
@@ -0,0 +1,73 @@
+const childProcess = require("child_process");
+
+class VSCodeDebugger {
+  constructor(config, txHash) {
+    this.config = config;
+    this.txHash = txHash;
+  }
+
+  /**
+   * This function is responsible for opening the debugger in vscode.
+   */
+  async run() {
+    // Sets the URL
+    const url = new URL("/debug", "vscode://trufflesuite-csi.truffle-vscode");
+
+    const disableFetchExternal = this.config.fetchExternal ? false : true;
+
+    // Sets the query parameters
+    url.searchParams.set("txHash", this.txHash);
+    url.searchParams.set("workingDirectory", this.config.working_directory);
+    url.searchParams.set("providerUrl", this.getProviderUrl());
+    url.searchParams.set("network", this.config.network);
+    url.searchParams.set("disableFetchExternal", disableFetchExternal);
+
+    // Opens VSCode based on OS
+    const openCommand = process.platform === "win32" ? `start ""` : `open`;
```

Comment:
Does `open` also works in Linux?

Reply:
IIRC, `win32: start`, `osX: open`, and *nix is `xdg-open or similar resource openers mimeopen, mimeo, linopen, handlr etc`.